### ISSUE TITLE
Is it possible to make front and rear output devices playback two different audio streams simultaneously?

### ISSUE THREAD

**Issue by Leomund** [pal1000/Realtek-UAD-generic#4]
Hi. In the old Realtek HD Audio Manager  it was possible to "Make front and rear output devices playback two different audio streams simultaneously".
![1](https://user-images.githubusercontent.com/6814625/60078138-0802b700-9734-11e9-8007-a5c9a80ce6db.jpg)
This provided 2 separate audio playback devices for front and rear jacks and possibility to switch them from windows. Also, this is how the hd audio driver, provided by Microsoft Windows 10, behaves by default.
Is it possible with UAD driver and Realtek Audio Console?

**Comment by pal1000** [comment]
I never used multiple audio output devices so I can't be of much assistance.  I don't know what features Realtek Audio console provides in this scenario.

**Comment by Leomund** [comment]
I just found this screenshot on the web. And there is this feature that I missing. Both of them: multiple audio output devices and ui for connectors retasking too. But this screenshot is from ASUS Realtek Audio Console. Is it just branded version of Realtek Audio Console, or some sort of add-on above it? Can someone check is there this features in generic Realtek Audio Console?

**Comment by shinoamakusa** [comment]
I have Asus Realtek chip but using generic console:
![image](https://user-images.githubusercontent.com/9992759/62380221-9f121a00-b4fd-11e9-977c-00cc42a0f0d9.png)

**Comment by Leomund** [comment]
@shinoamakusa 
Strange. What realtek audio chip do you have? I have Realtek ALC1220 and missing this feature, same as connector retasking feature, that you have. But i already saw other boards with same Realtek ALC1220, and they have this features. I am simply don't understand why the same hardware with same software have a different features available?

**Comment by shinoamakusa** [comment]
I have ALC887 I believe ... I got Asus P8Z77-V LX motherboard

**Comment by shinoamakusa** [comment]
But as you can see I have option to split up only Recording devices not Playback ones

**Comment by Leomund** [comment]
> But as you can see I have option to split up only Recording devices not Playback ones

Yeah, I see, but also I noticed that you have retasking feature at least for speaker out, and I am not.
![2019-08-02 18_45_40](https://user-images.githubusercontent.com/6814625/62382248-d6acb080-b555-11e9-83c8-97064b8519ba.png)

**Comment by shinoamakusa** [comment]
Not sure why ... but I wish retasking was available for more than just one output

**Comment by petdoskav** [comment]
Hi every one :D , retasking work fine with 775 gigagyte board , two streams can be achived by retasking a port on a realtek audio console or the panel ( see pictures above I wont send new ones they depend on your hda or uad driver ) , and then declaring one stream wasapi output , I'm using MediaMonkey ( free ) for wasapi and Kodi for 7.1 audio output to achive two streams ( hope this will be fixed in future with regular directmedia streams ). There is one problem that I found common with every driver, sound start to pup up like fire cracker when it gets amplitude saturated , this seem to be a problem for everyone with 775 board or alc889a. Please help or please explain this phenomena. Oh, to not forget about install : The best way to install this driver is to install uad 8501 ( automatically will uninstall previous driver and fix all problems ) then update in devices properties with this 8813 and then install the realtek audio console from Microsoft store if you have win 10. URL for the app :  https://www.microsoft.com/en-us/p/realtek-audio-control/9p2b8mcsvpln?activetab=pivot:overviewtab . Good job pal1000, every one have nice time with this release I sure do :D 👍

**Comment by petdoskav** [comment]
Hi again, about this : There is one problem that I found common with every driver , sound start to pup up like fire cracker when it gets amplitude saturated , this seem to be a problem for everyone with 775 board or alc889a. It seems that in speakers properties when enable audio enhancements check box is unchecked ( hit apply ) then checked again ( apply ) the driver property is reset and back to normal, i think that this is related with sound amplification options ( perhaps stacked up : extra stereo, better bass, better voice and etc )  . Can this thing be fixed in next release ?  Thanks again pal1000 👍 Good job :) ( forgive my english i'm not a native speaker )

**Comment by pal1000** [comment]
Driver bugs like this can only be tackled by Realtek, @petdoskav. I only do the packaging based on this recipe:
https://github.com/alanfox2000/realtek-universal-audio-driver/wiki/Installation#b-using-pnputil-only-for-user-who-have-hardware-id-is-not-found-error-when-using-universal-audio-driver-inf-editor

**Comment by mckennajones** [comment]
@shinoamakusa, I'm experiencing the same behavior, did you find a solution?

**Comment by shinoamakusa** [comment]
> 
> 
> @shinoamakusa, I'm experiencing the same behavior, did you find a solution?

Nope :(

**Comment by Hazbelll** [comment]
This might exist for you already, but under a new name, at least is the case for my laptop, so I thought I'd chime in here to let you guys know.

If you open the Realtek Audio Console and navigate to _"Device advanced settings"_, you might see a section called _"Audio Director"_ which presents two options to choose from: _"Classic mode"_ or _"Multi-stream mode"_. It says you must restart for the change to take effect, but in my case it works immediately.

If you want to separate your Inputs/Outputs, you will want to select **"Multi-stream mode"**. In my case, it allows for independent playback/recording of both the laptop's internal speakers and microphone and the External Headphones/Microphone combo jack together.

Audio Director and these options exist in the classic Realtek Control Panel too, but I assume like everything else to do with the Realtek Audio Packages, it is totally dependent on your specific codec/driver combo whether it will be visible/functional or not. Looking at the screenshots above, it seems you are unfortunately out of luck and I feel this variation might only exist on laptops.

**Comment by epman100** [comment]
the multi-stream option to "Make front and rear output devices playback two different audio streams simultaneously" is either hidden or unavailable, especially for ALC887 audio chipsets.

I know because my friend's old Asus M5A78L-M/LE USB3 board uses ALC887 (ID on there is "DEV_0887&SUBSYS_10438576") and that option never appears whether using legacy HDA or UAD drivers.  I had to create a specialized configuration file (rtkhdaud.dat) w/ a PBIndependentSupport feature enabled from an undocumented Realtek HDA driver test tool (ask alanfox2000 about that tool) and place that config file onto the Windows\System32\drivers folder and reboot in order for that option to appear in either the Realtek HD Audio Manager app or in the Realtek Audio Console

The rtkhdaud.dat config trick to have the multi-stream option seems to work with ALC8xx audio chipsets

**Comment by Leomund** [comment]
@epman100 it worked! The 2nd audio output is working separately. But playback device switch has not appeared in device advanced settings. Maybe you know how to deal with it? And also may be this tool can help with [enabling connector retasking](https://github.com/pal1000/Realtek-UAD-generic/issues/3#issue-460253104)?

**Comment by epman100** [comment]
I don't know, Leomund.
what motherboard do you have?

notice I said **ALC887**, not ALC1220 in my previous post.  I thought most ALC1220 chips on many boards would already have both the Playback Device and Recording Device advanced settings options available in the Realtek Audio Console; guess I was wrong.

**Comment by Leomund** [comment]
@epman100 My mistake. I was not sure that this problem is related to the motherboard. My is **ASRock z370 Extreme4**.

> I thought most ALC1220 chips on many boards would already have both the Playback Device and Recording Device advanced settings options available in the Realtek Audio Console like with ALC892

You right. On other motherboards with same ALC1220 these settings are available by default. But not on mine.

**Comment by Beauenheim** [comment]
Hey everyone, I had this exact same issue with my Asrock X570 Steel Legend, it also has an ALC1220.

I found a driver that restores the option!

Here is the driver that worked for me.

https://www.techpowerup.com/forums/threads/asus-realtek-drivers-from-2018-prepatched-for-ddl-and-dts-support-through-spdif.263768/
![56EyFIV](https://user-images.githubusercontent.com/6319423/78844441-fd639680-79c2-11ea-904c-d078fffcf8ac.png)

**Comment by danthephotographyman** [comment]
> My mistake. I was not sure that this problem is related to the motherboard. My is ASRock z370 Extreme4.

I have the same exact problem. Weirdly my PC was working jsut fine until i formatted and made a fresh windows install two days ago. I'm on that ASRock z370 Extreme4 as well :( 
No idea how to get the front and back audio seperated

**Comment by grimmute** [comment]
Unfortunately I have the same issue with my X470 Taichi (ALC1220).

Neither the UAD drivers from ASRock website nor the latest UAD drivers here has an option to separate front and rear audip jacks.

My only "solution" to this so far is not to install any sound drivers and let Windows high definition audio drivers handle it. At least this way I can separate headphones and speakers. Unfortunately with these drivers I have microphone echo problems... It would be nice to be able to separate the inputs with the generic UAD drivers so I can also use other options with it.

**Comment by NYCJames** [comment]
I also cannot find the option 
![image](https://user-images.githubusercontent.com/38173610/83952723-a7676f80-a808-11ea-9e88-aea66186a8d4.png)

**Comment by epman100** [comment]
> 
> 
> I also cannot find the option
> 

what is the Realtek ALCxxx model on your computer or motherboard, NYCJames?

**Comment by NYCJames** [comment]
> > I also cannot find the option
> 
> what is the Realtek ALCxxx model on your computer or motherboard, NYCJames?

It appears to be an ALC898

**Comment by epman100** [comment]
> 
> 
> > My mistake. I was not sure that this problem is related to the motherboard. My is ASRock z370 Extreme4.
> 
> I have the same exact problem. Weirdly my PC was working jsut fine until i formatted and made a fresh windows install two days ago. I'm on that ASRock z370 Extreme4 as well :(
> No idea how to get the front and back audio seperated



> 
> 
> Unfortunately I have the same issue with my X470 Taichi (ALC1220).
> 
> Neither the UAD drivers from ASRock website nor the latest UAD drivers here has an option to separate front and rear audio jacks.
> 
> My only "solution" to this so far is not to install any sound drivers and let Windows high definition audio drivers handle it. At least this way I can separate headphones and speakers. Unfortunately with these drivers I have microphone echo problems... It would be nice to be able to separate the inputs with the generic UAD drivers so I can also use other options with it.

@ danthephotographyman  & grimmute:

maybe ask AlanFinotty1995 on using the [Realtek HDA driver test tool](https://www.techpowerup.com/forums/threads/realtek-audio-device-tweak-raltek-hda-driver-test-utility-help-needed.263482/) to create an rtkhdaud.dat config file for your ALC1220.

it's unfortunate that ALC1220 chips don't get all the audio options like what I have on my Gigabyte board with Realtek ALC892 where I have both "advanced" playback and recording options available, including the ability to separate speaker & headphone outputs.

**Comment by epman100** [comment]
> 
> It appears to be an ALC898

I forgot to ask earlier, what computer or motherboard are you using?

**Comment by NYCJames** [comment]
> > It appears to be an ALC898
> 
> I forgot to ask earlier, what computer or motherboard are you using?

I am using an Alienware aurora r7

**Comment by Beauenheim** [comment]
Hi all, here is the fix!
Use the Realtek HDA driver test utility to change the following options:

![3di595f6zyx41](https://user-images.githubusercontent.com/6319423/84491557-f62a6480-ac61-11ea-96b3-837cf19ebe2a.png)
![mvgnj2f6zyx41](https://user-images.githubusercontent.com/6319423/84491576-fb87af00-ac61-11ea-8db9-063ac137151f.png)

Realtek HDA Driver test utility download:
https://www.techpowerup.com/forums/threads/realtek-audio-device-tweak-raltek-hda-driver-test-utility-help-needed.263482/post-4269698

Source:
https://www.techpowerup.com/forums/threads/the-ultimate-realtek-hd-audio-driver-mod-for-windows-10.232921/post-4261951

**Comment by Leomund** [comment]
@Beauenheim  this solution was already proposed [here ](https://github.com/pal1000/Realtek-UAD-generic/issues/4#issuecomment-590220225) by @epman100. 
[It works.](https://github.com/pal1000/Realtek-UAD-generic/issues/4#issuecomment-590958668) But not quite as it should. Playback device switch option has not appeared in device advanced settings. The 2nd audio output is working separately constantly.
![image](https://user-images.githubusercontent.com/6814625/84501758-bdb97380-acbf-11ea-96b8-e2544546e065.png)

**Comment by Beauenheim** [comment]
I understand, as epman's post pointed me in the right direction, but as this github is the top thing that comes up when you google search the issue of not being able to play both streams at the same time, I figured the pictures would help everyone else. 

I can confirm that in the realtek audio console, the option is still missing, but at least creating the .dat file works.

**Comment by grimmute** [comment]
> 
> 
> I understand, as epman's post pointed me in the right direction, but as this github is the top thing that comes up when you google search the issue of not being able to play both streams at the same time, I figured the pictures would help everyone else.
> 
> I can confirm that in the realtek audio console, the option is still missing, but at least creating the .dat file works.

Thanks a lot for sharing the links. I created the .dat file and installed the latest UAD drivers. Everything works fine right now. This issue was driving me crazy (as Microsoft drivers had echo on microphone).

**Comment by NYCJames** [comment]
Did you guys also get connector retasking working?

**Comment by Leomund** [comment]
@Beauenheim I am appreciate your clarification. Hope it helps other people. ;)
I just wanted to clarify again that this is only a partial solution to the problem and does not work quite as it should.

**Comment by Leomund** [comment]
> Did you guys also get connector retasking working?
Found absolutely no way to make this work.

> The HDA driver test utility is not giving me a separate playback either.
Does this not give you the “separate playback” switch in advanced settings, or a separate playback device at all?

**Comment by NYCJames** [comment]
> Does this not give you the “separate playback” switch in advanced settings, or a separate playback device at all?

No, it does not 
![image](https://user-images.githubusercontent.com/38173610/84530345-fa31a300-acb0-11ea-9906-5534aad9c600.png)

**Comment by Leomund** [comment]
May it be disabled in windows playback devices?
![image](https://user-images.githubusercontent.com/6814625/84530667-f8390580-aceb-11ea-8208-63af07ea43ed.png)

**Comment by NYCJames** [comment]
> Can it be disabled in windows playback devices?
I only have these 5 devices and 2 are realtek
![image](https://user-images.githubusercontent.com/38173610/84531027-e9356180-acb1-11ea-8c15-357300aa6a14.png)

**Comment by Leomund** [comment]
@NYCJames You may have missed something in the HDA driver test utility.
Or may be this is another oddity from Reatek. Their devices always working strange and unclear. The same devices work completely differently.

**Comment by NYCJames** [comment]
> @NYCJames You may have missed something in the HDA driver test utility.
> Or may be this is another oddity from Realtek. Their hardware and software always working strange and unclear. The same devices work completely different ways.

I just pressed Save after editing everything shown in the pictures above
![image](https://user-images.githubusercontent.com/38173610/84532048-8f359b80-acb3-11ea-9e65-a77ec7f02ebb.png)

**Comment by Beauenheim** [comment]
26 22 15 14 13 10 9 8 6 5 4 3 2 1 0

These are all the boxes I have checked in my jack config, yours might require more checked.

**Comment by NYCJames** [comment]
> @NYCJames 26 22 15 14 13 10 9 8 6 5 4 3 2 1 0
> 
> These are all the boxes I have checked in my jack config, yours might require more checked.

Alright I checked more, after saving do I just reboot?

**Comment by Beauenheim** [comment]
Any time you make changes in that config, a reboot is recommended. If this causes any problems, make sure to delete your .dat file as the standard uninstaller didn't seem to grab it for me.

**Comment by NYCJames** [comment]
> Any time you make changes in that config, a reboot is recommended. If this causes any problems, make sure to delete your .dat file if starting over and reinstalling as the standard uninstaller didn't seem to grab it for me.

Now I have a separate mic, but not a separate headphones/speaker
![image](https://user-images.githubusercontent.com/38173610/84533532-47fcda00-acb6-11ea-8e80-ed2e84367dee.png)

**Comment by Leomund** [comment]
@NYCJames and you also got a switch. Very strange. When I get a separate device, I did not receive such switch for it.
![image](https://user-images.githubusercontent.com/6814625/84534740-279f4080-acf3-11ea-9fce-50f221d8b875.png)

**Comment by NYCJames** [comment]
> @NYCJames and you also got a switch. Very strange. When I get a separate device, I did not receive such switch for it.

What switch did I get? I see we both only have the mic separation switch.

**Comment by Leomund** [comment]
@NYCJames Yes. But you didn't have it before, right? You received it along with the activation of a separate recording device. And when I received the separate playback device, such a switch for it did not appear.

**Comment by Beauenheim** [comment]
Just to really confuse you, I seem to have connector retasking in my options.

![vva8VRD](https://user-images.githubusercontent.com/6319423/84535285-c5b9e900-aca8-11ea-90d6-daa261f05c2d.png)

**Comment by Leomund** [comment]
@Beauenheim Wow. You received it through the driver test utility. Or did you have them initially?

**Comment by NYCJames** [comment]
@Leomund Yes. I just got the switch for mic separation.

**Comment by Leomund** [comment]
@NYCJames And it's weird. I did not get such a switch to separate playback. My separate playback is always enabled, I can't control it.

**Comment by Beauenheim** [comment]
Retasking came default with my driver it seems.

If you use the FindPolicyText option in the bottom left section of the Realtek HDA driver test utility, and enter the word: task

It shows multiple options related to tasking and retasking, maybe it could help if you went through and turned some of those on?

**Comment by Leomund** [comment]
@Beauenheim Yes, I saw it. And I tried to experiment with it. But I still couldn't get retasking to work.

**Comment by hori87** [comment]
Thanks.  I could fix the same problem with Realtek HDA driver test utility.
Initially there was no playback device switch, but now I can "make front and rear output devices playback two different audio streams simultaneously".

ACL1200 on ASRock X570 Pro4

**Comment by epman100** [comment]
> 
> 
> @Beauenheim Yes, I saw it. And I tried to experiment with it. But I still couldn't get retasking to work in any way.

then most likely the "retasking" feature is not supported for your motherboard

**Comment by mirh** [comment]
How are these drivers handling retasking, when I thought this had been blocked after 6.0.1.8000?

**Comment by jackblk** [comment]
> > @NYCJames You may have missed something in the HDA driver test utility.
> > Or may be this is another oddity from Realtek. Their hardware and software always working strange and unclear. The same devices work completely different ways.
> 
> I just pressed Save after editing everything shown in the pictures above
> ![image](https://user-images.githubusercontent.com/38173610/84532048-8f359b80-acb3-11ea-9e65-a77ec7f02ebb.png)

This works for me!

I'm using ALC1150, from ASUS B85M-GAMER motherboard. I don't have this setting, which i'm craving. Thankfully this helped me!

After using this method, the Realtek UI changed from ASUS to generic Realtek, and it has the option 👍

**Comment by jackblk** [comment]
> Hi all, here is the fix!
> Use the Realtek HDA driver test utility to change the following options:
> 
> ![3di595f6zyx41](https://user-images.githubusercontent.com/6319423/84491557-f62a6480-ac61-11ea-96b3-837cf19ebe2a.png)
> ![mvgnj2f6zyx41](https://user-images.githubusercontent.com/6319423/84491576-fb87af00-ac61-11ea-8db9-063ac137151f.png)
> 
> Realtek HDA Driver test utility download:
> https://www.techpowerup.com/forums/threads/realtek-audio-device-tweak-raltek-hda-driver-test-utility-help-needed.263482/post-4269698
> 
> Source:
> https://www.techpowerup.com/forums/threads/the-ultimate-realtek-hd-audio-driver-mod-for-windows-10.232921/post-4261951

This works for me!

I'm using ALC1150, from ASUS B85M-GAMER motherboard, 6.0.1.7899. I don't have this setting, which i'm craving. Thankfully this helped me!

After using this method, the Realtek UI changed from ASUS to generic Realtek, and it has the option 👍

**Comment by KKire** [comment]
Has anyone found any reliable solution to fix this?
I have Asus TUF FX504GD laptop with ALC3236-VB2 codec and tries all the above settings to change in Realtek HDA Driver test utility and I'm not able to separate headphones from speakers. 
I have tried almost any driver out there, from Asus website, WU, this generic one, UAD from rog website (ASUS TUF NB version) and none of them worked.
If I tick 26 option, it shows up headphones but it doesn't produce any sound and speakers option is still for both.

The only option to work is the generic one from Microsoft (High Definition Audio Device) but with it the microphone is not detected from the headphones and I need that microphone to work.

**Comment by msa48** [comment]
The utility helped me too with drivers (6.09.9038.1) from ASRock site.  I have an ASRock B450 Pro4-F motherboard
I was wounder when I didn't see a split output stream feature when installed official drivers. In my ten years old motherboard it worked like a charm whitout any magic.

@FraCart, R2.82 Realtek driver is buggy. In an old R2.81 (6.0.1.8036) manager is working. But it doesn't show split button too. And with this utility a second output was unconnected.

I also tried to remove all drivers and use MS default (10.0.19041.264). It also work. All input and output streams was splitted and activated by default. But an input device `Stereo Mixer` was gone, as an any manager console.

**Comment by Bibabutzemann123** [comment]
@FraCart this would be a fine solution for me if i there would still be an option to eliminate Echo when using microphone...

@msa48 did u find a solution yet? I mean it cant be a hardware limitation if it works with MS defaulot drivers.... This is really frustrating to me.

**Comment by msa48** [comment]
@Bibabutzemann123, yes. I got two working outputs by this
https://github.com/pal1000/Realtek-UAD-generic/issues/4#issuecomment-643190347

**Comment by Bibabutzemann123** [comment]
@msa48 oh now it works for me too.
I tried that first, but it didnt work.
I think this time i also clicked on save in the utility..

**Comment by notstax** [comment]
> Hi all, here is the fix! Use the Realtek HDA driver test utility to change the following options:
> 
> ![3di595f6zyx41](https://user-images.githubusercontent.com/6319423/84491557-f62a6480-ac61-11ea-96b3-837cf19ebe2a.png) ![mvgnj2f6zyx41](https://user-images.githubusercontent.com/6319423/84491576-fb87af00-ac61-11ea-8db9-063ac137151f.png)
> 
> Realtek HDA Driver test utility download: https://www.techpowerup.com/forums/threads/realtek-audio-device-tweak-raltek-hda-driver-test-utility-help-needed.263482/post-4269698
> 
> Source: https://www.techpowerup.com/forums/threads/the-ultimate-realtek-hd-audio-driver-mod-for-windows-10.232921/post-4261951

This worked for me.
Mobo: Asus A320M Pro4
Codec: ALC892

**Comment by notstax** [comment]
> Realtek HDA Driver test utility download: https://www.techpowerup.com/forums/threads/realtek-audio-device-tweak-raltek-hda-driver-test-utility-help-needed.263482/post-4269698
> 
> Source: https://www.techpowerup.com/forums/threads/the-ultimate-realtek-hd-audio-driver-mod-for-windows-10.232921/post-4261951

This worked for me.
Mobo: Asus A320M Pro4
Audio Codec: ALC892
Realtek Driver Version: 6.0.1.8403

Thank you!!

**Comment by drLeChapeau** [comment]
I know this is quite an old topic, but I would like to share with you guys; This fixed my output problem as well on Windows 11 with:

Mobo: ASRock AB350 PRO4
Audio Codec: ALC892
Realtek Driver Version: 6.0.9231.1

I have been looking for a solution to switch between my front and back audio inputs for so long. I know have an extra Output device to select so I can switch between my speakers and my headset.

Thanks a LOT!

**Comment by ispekupirog** [comment]
@Leomund Hey! I found a solution to unhide «Playback Device» settings.
Just Uncheck number 24 (NoShowPBMultiStreaming).
Don't forget to reboot your computer.

![28042022](https://user-images.githubusercontent.com/58360795/165692855-2b7e336a-ed87-4984-bde9-8b0b3c880bda.png)

**Comment by Kuiriel** [comment]
Sorry for having made another thread instead of coming back to this gem.

Is there anything I should be cautious of if attempting this solution on newer motherboards like the z790 Apex? I followed the above solutions on the Maximus XI Hero early last year and was able to get it working, much to my delight. I'm a little wary of breaking things when there may be new bugs in the newer boards and codecs - I used to have a Xonar Essence STX soundcard which I thought was oh so fancy, yet it repeatedly blew out my ear drums with high pitched screeches at complete random.

**Comment by Kuiriel** [comment]
If anyone ever figures out how to use RTKConfig or some similar tool with Realtek USB Audio Devices like on the newer motherboards, I would absolutely love to hear about it. When I open RTKConfig and click Driver Policy, nothing is ticked whatsoever.

**Comment by Kyogre** [comment]
ispekupirog, thank you! It helped for ASRock motherboard.
Notice: it is important to not change anything in "Customized" tab. Change only what is told in [ispekupirog's comment](https://github.com/pal1000/Realtek-UAD-generic/issues/4#issuecomment-1111809760).

You can additionally try installing custom drivers from https://github.com/pal1000/Realtek-UAD-generic/ but I suppose ispekupirog's method works even without it.

**Comment by lo2lo2a82** [comment]
> The utility helped me too with drivers (6.09.9038.1) from ASRock site. I have an ASRock B450 Pro4-F motherboard I was wounder when I didn't see a split output stream feature when installed official drivers. In my ten years old motherboard it worked like a charm whitout any magic.
> 
> @FraCart, R2.82 Realtek driver is buggy. In an old R2.81 (6.0.1.8036) manager is working. But it doesn't show split button too. And with this utility a second output was unconnected.
> 
> I also tried to remove all drivers and use MS default (10.0.19041.264). It also work. All input and output streams was splitted and activated by default. But an input device `Stereo Mixer` was gone, as an any manager console.

simply removing the realtek installation made things worked for me .. thank you for the suggestion

**Comment by Lemonwump** [comment]
My situation is even more dire than what i've seen here.
![Screenshot 2023-08-10 153009](https://github.com/pal1000/Realtek-UAD-generic/assets/141963631/5523b0f5-a890-465f-8e99-3386a0c68f2e)
(msinfo32)System Model: B550 Phantom Gaming 4/ac
(Realtek)Audio Driver Version: 6.0.9030.1
(Realtek)UI Version: 1.1.137.0

I wanted to make front and rear output devices playback two different audio streams simultaneously and repurpose the audio jacks but apparently i cannot. Someone pls help.

**Comment by epman100** [comment]
> Sorry for having made another thread instead of coming back to this gem.
> 
> Is there anything I should be cautious of if attempting this solution on newer motherboards like the z790 Apex? I followed the above solutions on the Maximus XI Hero early last year and was able to get it working, much to my delight. I'm a little wary of breaking things when there may be new bugs in the newer boards and codecs - I used to have a Xonar Essence STX soundcard which I thought was oh so fancy, yet it repeatedly blew out my ear drums with high pitched screeches at complete random.

the Asus ROG Maximus XI Hero board uses Realtek ALC1220 (S1220) HDA chipset as noted in it's [tech specs page](https://rog.asus.com/motherboards/rog-maximus/rog-maximus-xi-hero-model/spec/).  while the ROG Maximus Z790 Apex uses ALC4080 (an entirely different chipset than ALC12xx) as noted in the [tech specs page](https://rog.asus.com/motherboards/rog-maximus/rog-maximus-z790-apex-model/spec/).

> If anyone ever figures out how to use RTKConfig or some similar tool with Realtek USB Audio Devices like on the newer motherboards, I would absolutely love to hear about it. When I open RTKConfig and click Driver Policy, nothing is ticked whatsoever.

that's because the undocumented Realtek HDA driver test tool was designed only for "standard" HD audio ALC2xx/6xx/8xx/1xxx chipsets.  it was never designed to work on Realtek USB audio chipsets like ALC4050/ALC4080/ALC4082 and Realtek never produced an equivalent tool for the less-common usb audio hardware.
so you need to get a different board that does NOT use ALC408x.  I never liked (and despised) any boards with Realtek ALC40xx usb audio hardware due to other various problems with them.

**Comment by epman100** [comment]
> (msinfo32)System Model: B550 Phantom Gaming 4/ac (Realtek)Audio Driver Version: 6.0.9030.1 (Realtek)UI Version: 1.1.137.0
> 
> I wanted to make front and rear output devices playback two different audio streams simultaneously and repurpose the audio jacks but apparently i cannot. Someone pls help.


did you follow what [ispekupirog has said](https://github.com/pal1000/Realtek-UAD-generic/issues/4#issuecomment-1111809760)?

**Comment by kubapl2004gmail** [comment]
Don't forget to turn on Use Config Hard Coding in General - General Setting on Realtek HDA driver test utility, ok?
